src/dmd/identifier.d

```diff
@@ -30,17 +30,24 @@ extern (C++) final class Identifier : RootObject
 {
 private:
     const int value;
-    const char* name;
-    const size_t len;
+    const char[] name;
 
 public:
+    /**
+       Construct an identifier from a D slice
+
+       Note: Since `name` needs to be `\0` terminated for `toChars`,
+       no slice overload is provided yet.
 
+       Params:
+         name = the identifier name
+                There must be `\0` a termination at name[$].
```

Comment:
Can an assert be added to verify this?

Reply:
Nope, since it's out of bounds

Reply:
So how can the C string returned by `toChars` be used?

Reply:
Oh right, this is a pointer 🤦‍♂️ 
So we could add it, indeed.

However I'm a bit wary of doing so though, as this might quickly proliferate all over the place. In fact there is no precedent for it.
This is just adding documentation, not changing this part of the code. If anything, it reduces the dependency on `\0`.

Reply:
ok